kiota_abstractions/serialization/untyped_dict.py

```diff
@@ -0,0 +1,24 @@
+from .untyped_node import UntypedNode
+
+class UntypedDictionary(UntypedNode):
+    """
+    Represents an untyped node with dictionary value.
+    """
+
+    def __init__(self, value: dict) -> None:
+        """
+        Creates a new instance of UntypedDictionary.
+
+        Args:
+            value (dict): The key-value pair associated with the node.
+        """
+        self.__value = value
+
+    def get_value(self) -> dict:
```

Comment:
is it possible to constrain the type to a dictionary of [key, UntypedNode] ?

Reply:
Yes, will [str, UntypedNode] work? 
`def __init__(self, value: dict[str:UntypedNode]) -> None:`
`def get_value(self) -> dict[str:UntypedNode]:`

Reply:
Yeah. Constraining with the base should allow the values to be any of the derived types.

Reply:
I have one more concern with regards to constraining the type of list and dictionary or any other argument type of python function. Python doesn't raises an exception if the argument passed is not of the same type. If we want to handle that, we should use try-except and assert the type. e.g. https://onlinegdb.com/u3nKT5i0M
Same goes with return type as well. These are just type hints and will not raise errors if the provided types do not match the actual values. They don't enforce type-safety.

I will implement something like that provided in the example if you like or else will just write type-hints.

thanks

Reply:
Ok Cool! I will implement something to raise the exception other than type hints.